bson/bsonrw/value_writer.go

```diff
@@ -612,5 +611,21 @@ func (vw *valueWriter) writeLength() error {
 }
 
 func isValidCString(cs string) bool {
-	return !strings.ContainsRune(cs, '\x00')
+	return strings.IndexByte(cs, 0) == -1
```

Comment:
Real reason is that `'\x00'` is a byte (it equals `0` not `00000000`) and since it is less that MaxRune (`0x80`) it just calls `IndexByte` - this change just omits an unnecessary function call (and one that can't currently be inlined because its complexity score exceeds what is allowed for inlining).

Reply:
I think we're both saying the same thing. `0x80` (hex) == `10000000` (binary). I can update the comment with a reference to the corresponding code in `strings.ContainsRune` for additional clarity.

Reply:
Ah, I thought you meant `10000000` as hex or a 32bit rune here since that's what IndexRune takes.